mlflow/entities/run_info.py

```diff
@@ -81,13 +83,17 @@ def __eq__(self, other):
             return self.__dict__ == other.__dict__
         return False
 
-    def _copy_with_overrides(self, status=None, end_time=None, lifecycle_stage=None):
+    def _copy_with_overrides(
+        self, status=None, end_time=None, delete_time=None, lifecycle_stage=None
+    ):
         """A copy of the RunInfo with certain attributes modified."""
         proto = self.to_proto()
         if status:
             proto.status = status
         if end_time:
             proto.end_time = end_time
+        if delete_time or delete_time == 0:
+            proto.delete_time = delete_time
```

Comment:
If we adopt the proposal from https://github.com/mlflow/mlflow/pull/6354/files#r933551129, I think we can avoid having to expose the deletion time on the RunInfo.

Reply:
The run_info is used by the file_store's `delete_run` function, where delete_time will be updated into the file store. if we remove this line how do I get the delete_time information on file_store repository?

Reply:
I realised that without changes to `RunInfo` and the protos, the `--older-than` flag would not work for file-stores. Do you want this feature to only work with sqlalchemy databases?